pkg/server/serverpb/status.proto

```diff
@@ -636,6 +636,30 @@ message CancelQueryResponse {
   string error = 2;
 }
 
+// Request object for issuing a pgwire query cancel request.
+message PGWireCancelQueryRequest {
+  // ID of gateway node for the query to be canceled.
+  //
+  // TODO(rafi): use [(gogoproto.customname) = "NodeID"] below. Need to
+  // figure out how to teach grpc-gateway about custom names. Refer to
+  // https://github.com/gogo/protobuf/issues/331.
+  //
+  // node_id is a string so that "local" can be used to specify that no
+  // forwarding is necessary.
+  string node_id = 1;
+  // "Secret" ID of query to be canceled. This comes from the BackendKeyData
+  // generated during the pgwrite protocol flow.
+  uint32 secret_id = 2 [(gogoproto.customname) = "SecretID" ];
+}
+
+// Response returned by target query's gateway node for a pgwire cancel request.
+message PGWireCancelQueryResponse {
+  // Whether the cancellation request succeeded and the query was canceled.
+  bool canceled = 1;
```

Comment:
super nit: i think it's more common to have something called `success`, but up to you. 
would it be more sane to have this have no response but a grpc error if this fails?

Reply:
Makes sense to me, but I was matching the response format of the non-pgwire CancelQueryResponse. I lean in favor of matching that

Reply:
mm, up to you.